molviewspec/molviewspec/nodes.py

```diff
@@ -39,17 +39,30 @@
 ]
 
 
+CustomT = Optional[Mapping[str, Any]]
+RefT = Optional[str]
+
+
 class Node(BaseModel):
     """
     Base impl of all state tree nodes.
     """
 
     kind: KindT = Field(description="The type of this node.")
     params: Optional[Mapping[str, Any]] = Field(description="Optional params that are needed for this node.")
-    additional_properties: Optional[Mapping[str, Any]] = Field(
-        description="Optional free-style dict with custom, non-schema props."
-    )
     children: Optional[list[Node]] = Field(description="Optional collection of nested child nodes.")
+    custom: CustomT = Field(description="Custom data to store attached to this node.")
+    ref: RefT = Field(description="Optional reference that can be used to access this node.")
```

Comment:
Wouldn't it be simpler to just include "custom" and "ref" as one of the node params? It would be defined in a params base class that all param classes would extend.

Instead of for example:
```
class ComponentInlineParams(BaseModel):
    selector: Union[ComponentSelectorT, ComponentExpression, list[ComponentExpression]] = Field(...)
```

We would have:
```
class BaseParams(BaseModel):
    custom: ...
    ref: ...
class ComponentInlineParams(BaseParams):
    selector: Union[ComponentSelectorT, ComponentExpression, list[ComponentExpression]] = Field(...)
```

I believe this will simplify code in `Node.__init__` and `make_params` as they won't need to treat "custom" and "ref" in any special way.

Reply:
What would be the benefit of this?

Reply:
Although, you are right, this would also allow to not have ref/custom on all nodes for example if we ever don't want to support it everywhere.